Source/WTF/wtf/threads/Signals.cpp

```diff
@@ -203,6 +206,157 @@ kern_return_t catch_mach_exception_raise_state_identity(mach_port_t, mach_port_t
     return KERN_FAILURE;
 }
 
+#ifdef EXCEPTION_IDENTITY_PROTECTED
+
+static thread_act_t threadIDGetThreadPort(uint64_t threadID)
+{
+    kern_return_t kr;
+    mach_msg_type_number_t i, thcnt;
+    thread_act_array_t threads;
+    struct thread_identifier_info tiinfo;
+    unsigned infoCount = THREAD_IDENTIFIER_INFO_COUNT;
+    thread_act_t thread = THREAD_NULL;
+
+    task_threads(mach_task_self(), &threads, &thcnt);
+    for (i = 0; i < thcnt; i++) {
+        if (thread != THREAD_NULL) {
+            // clean up all the unnecessary thread ports like a good citizen
+            ASSERT(thread != threads[i]);
+            (void)mach_port_deallocate(mach_task_self(), threads[i]);
+            continue;
+        }
+
+        kr = thread_info(threads[i], THREAD_IDENTIFIER_INFO,
+        (thread_info_t) &tiinfo, &infoCount);
```

Comment:
This should be `reinterpret_cast<thread_info_t>(& threadIdentifierInfo)` per WebKit style.